oasislmf/model_preparation/sum_inputs.py

```diff
@@ -297,110 +234,92 @@ def write_xref_file(summary_xref_df, target_dir):
     return summary_xref_fp
 
 
+def get_summary_xref_df(map_df, exposure_df, summaries_info_dict):
+    set_size = len(map_df)
+    summaryxref_df = pd.DataFrame()
+
+    # Infer il / gul xref type based on 'map_df'
+    if 'output_id' in map_df:
+        ids_set_df = map_df[['output_id']].rename(columns={"output_id": "output"})
+    else:
+        ids_set_df = map_df[['coverage_id']]
+
+    xref_size = len(map_df)
+    summaryxref_df = pd.DataFrame()
+
+    # For each granularity build a set grouping
+    for summary_set in summaries_info_dict:
+        summary_set_df = ids_set_df
+        cols_group_by = get_column_selection(summary_set)
+
+        if isinstance(cols_group_by, list):
+            summary_set_df['summary_id'] = group_by_oed(map_df, exposure_df, cols_group_by)
+        else:
+            # Fall back to setting all in single group
+            summary_set_df['summary_id'] =  pd.Series(1, index=range(set_size), dtype='uint32')
+
+        # Appends summary set to '__summaryxref.csv'
+        summary_set_df['summaryset_id'] =  pd.Series(summary_set['id'], index=range(set_size), dtype='uint32')
+        summaryxref_df = pd.concat([summaryxref_df, summary_set_df])
```

Comment:
In this `concat` statement you may want to pass the optional arguments `sort=True` (sorting the non-concatenation axis) and `ignore_index=True` (telling to create a new index starting from 0 on the concatenated dataframe, and not to just concatenate the two axes). Not setting `sort=True` will generate an annoying warning for users.